exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/aggregate/HashAggTemplate.java

```diff
@@ -325,10 +325,13 @@ public AggOutcome doWork() {
                 if (EXTRA_DEBUG_1) {
                   logger.debug("Received new schema.  Batch has {} records.", incoming.getRecordCount());
                 }
-                newSchema = true;
-                this.cleanup();
-                // TODO: new schema case needs to be handled appropriately
-                return AggOutcome.UPDATE_AGGREGATOR;
+                final BatchSchema newIncomingSchema = incoming.getSchema();
+                if ((! newIncomingSchema.equals(schema)) && schema != null) {
```

Comment:
You are quelling a new schema here without correctly reloading vector references. A schema change means: the schema has changed and/or the vectors have changed. Your check here isn't enough to guarantee correct behavior. It is possible that the schema is equal but the vectors changed.